src/lib.rs

```diff
@@ -332,7 +332,16 @@ where
     }
 
     fn poll_flush(mut self: Pin<&mut Self>, cx: &mut Context<'_>) -> Poll<Result<(), Self::Error>> {
-        (*self).with_context(Some((ContextWaker::Write, cx)), |s| cvt(s.write_pending()))
+        (*self).with_context(Some((ContextWaker::Write, cx)), |s| cvt(s.write_pending())).map(
+            |result| match result {
+                // It's not an error for the WebSocket to have closed normally.
+                // Note: `tungstenite` will only return this if a close frame has been sent, and
+                // doesn't allow sending anything else after that, so this can't arise when trying
+                // to flush something other than a close frame.
+                Err(::tungstenite::Error::ConnectionClosed) => Ok(()),
```

Comment:
```suggestion
                Err(::tungstenite::Error::ConnectionClosed) if self.closing => Ok(()),
```

If the connection is closed at another time while flushing then that seems problematic. Flushing doesn't only happen when closing.

Reply:
It doesn't seem like `self.closing` gets set in all the cases where the `WebSocket` gets closed - it's set in `poll_close`, but not when a close message is sent or recieved. (The case where a close message is received is the case where this situation should actually occur, since we're flushing the responding close frame automatically sent by `tungstenite`.)

Reply:
I guess that needs some updating too then

Reply:
The semantics of `closing` was that it was set only in cases when the user calls `poll_close()` for the following reason: our initial implementation (written years ago) did only one thing when `poll_close()` was called: it essentially added the `Close` message to the queue and tried to write it but it did not ensure that the `Close` frame is flushed (is actually sent to the peer). [There was a PR filed back then](https://github.com/snapview/tokio-tungstenite/pull/215) that fixed this behavior by making sure that once `poll_close()` is called, we send the `Close` frame and the subsequent calls to the `poll_close()` ensure that we're just flushing it ensuring that the queue has been flushed.

In other words to trigger the behavior introduced in this commit, one must first call `poll_close()`, get `Pending`, and then call `poll_flush()` (I don't know if it's the proper scenario? I assumed that normally when one calls `poll_close()` and it's not ready, one is supposed to call it until it's closed). I'm not sure if calling the `poll_flush()` after `poll_close()` is a good idea (is it?).

In other words: if you never call `poll_close()`, you'll never reach the branch of the code that this PR changes.